### DISCUSSION TITLE
Indicator result decimal? or double?

### DISCUSSION THREAD

**Discussion by KlekotOliwa** [DaveSkender/Stock.Indicators#1244] [Dev notes]
I have a question, most result type returns double? values, however AtrStopResult returns decimal? values

public sealed class AtrStopResult : ResultBase
{
public AtrStopResult(DateTime date)
{
Date = date;
}

public decimal? AtrStop { get; set; }
public decimal? BuyStop { get; set; }
public decimal? SellStop { get; set; }
}

Is this intentional or a mistake?

**Comment by DaveSkender** [comment]
Yes, intentional.  Over time we’ve reduced the use of `decimal`, where it is inconsequential, however we have kept it in places where the results need to reflect accurate price points.  Renko, HeikinAshi, ATR-Stop are a few examples.  More info and history:

- #649
